control_protocol.md

```diff
@@ -0,0 +1,359 @@
+:::
+
+
+#### Coordinator coordination
+
+Each Coordinator shall keep an up to date global {ref}`directory` with the names of all Components in the Network.
+For this, Coordinators shall tell each other regarding signing in and signing out Components and Coordinators.
+Coordinators shall send on request the Names of their local directory, or of their global directory, depending on the request type.
+
+For the format of the Messages, see {ref}`message-layer`.
+
+
+##### Coordinator sign in
+
+A Coordinator `Co1`joining a network follows a few steps:
+1. It signs in to one Coordinator `Co2` of the Network.
+2. It sends a CO_TELL_ALL message to `Co2`, to tell all other Coordinators about `Co1`s address.
+3. `Co2` tells all the Coordinators signed in (`Co3`, `Co4`...) about `Co1` with a CO_NEW message.
+4. These other Coordinators (`Co3`, `Co4`...) sign in to `Co1`.
+5. All Coordinators are connected to all others.
+
+Two Coordinators shall follow a more thorough sign in/sign out procedure, than Components (address is for example host an port).
+The sign in might happen because of a CO_NEW message arrived or at startup.
+The sign out might happen because the Coordinator shuts down.
+
+:::{mermaid}
+sequenceDiagram
+    participant r1 as ROUTER
+    participant d1 as DEALER
+    participant r2 as ROUTER
+    participant d2 as DEALER
+    Note over r1,d1: N1 Coordinator<br>at address1
+    Note over r2,d2: N2 Coordinator<br>at address2
+    Note over r1,d2: Sign in between two Coordinators
+    Note right of r1: shall connect<br>to address2
+    activate d1
+    Note left of d1: created with<br> name "temp-NS"
+    d1-->>r2: connect to address2
+    d1->>r2: CO_SIGNIN<br>N1, address1,<br>ref:temp-NS
+    par
+        d1->>r2: GET local directory
+    and
+        Note right of r2: stores N1 identity
+        activate d2
+        Note left of d2: created with<br>name "N1"
+        d2-->>r1: connect to address1
+        d2->>r1: CO_SIGNIN<br>N2, address2<br>your ref:temp-NS
+        Note right of r1: stores N2 identity
+        Note left of d1: name changed<br>from "temp-NS"<br>to "N2"
+        d2->>r1: GET local directory
+    end
+    d2->>r1: Here is my<br>local directory
+    Note right of r1: Updates<br>global directory
+    d1->>r2: Here is my<br>local directory
+    Note right of r2: Updates<br>global directory
+    Note over r1,d2: Sign out between two Coordinators
+    Note right of r1: shall sign out from N2
+    d1->>r2: CO_SIGNOUT
+    Note right of r2: removes N1 identity
+    d2->>-r1: CO_SIGNOUT
+    Note right of r1: removes N2 identity
+    deactivate d1
+:::
+
+
+##### Coordinator updates
+
+Whenever a Component signs in to or out of its Coordinator, the Coordinator shall send a note regarding this event to all the other Coordinators.
+The note shall contain the full name of the Component and the event type (sign in or out)
+The other Coordinators shall update their global directory according to this note (add or remove an entry).
```

Comment:
```suggestion
Whenever a Component signs in to or out of its Coordinator, the Coordinator shall send a message regarding this event to all the other Coordinators.
The message shall contain the full name of the Component and the event type (sign in or out)
The other Coordinators shall update their global directory according to this message (add or remove an entry).
```
I don't think we need to introduce the term "note" in addition to "message". Might need a specific message type.